tests/unit/command/test_external.py

```diff
@@ -0,0 +1,20 @@
+import os
+import shutil
+import stat
+import sys
+
+from dvc.main import main
+
+
+def test_external_cmd(tmp_dir, monkeypatch):
+    ext = tmp_dir / "dvc-ext"
+    ext.write_text(
+        f"#!{sys.executable}\nimport sys; print(sys.argv); sys.exit(101)"
+    )
+    ext.chmod(ext.stat().st_mode | stat.S_IEXEC)
+
+    path_value = os.pathsep.join([str(tmp_dir), os.getenv("PATH")])
+    monkeypatch.setenv("PATH", path_value)
+    assert shutil.which("dvc-ext") == str(ext)
+
+    assert main(["ext", "target", "--out", "out"]) == 101
```

Comment:
shouldn't the output (`print(sys.argv)`) be tested explicitly too?

Reply:
Done. Thanks.